src/components/TokenForm.js

```diff
@@ -0,0 +1,54 @@
+import React, { useState } from 'react';
+import { db } from '../lib/firebase';
+import Error from './Error';
+
+const TokenForm = ({ setToken }) => {
+  const [tokenInput, setTokenInput] = useState('');
+  const [hidden, setHidden] = useState(true);
+
```

Comment:
We tried using the react-firestore-hook `useCollection` here, but ran into problems with the `tokenInput` having no value initially, which would have required us to do something like :
```
  const [value, loading, error] = useCollection(
    tokenInput ? db.collection(tokenInput) : db.collection('novalue'),
    {
      snapshotListenOptions: { includeMetadataChanges: true },
    },
  );
```
which uses a placeholder value for the collection until inputToken is set to a real collection which seemed like a very clunky way to use the custom hook.

Reply:
Yeah, I agree that that looks way clunkier. Personally, I like the interface for hook that y'all made way better!
If you were super determined to put to use the react-firebase-hooks library, you could shorten the above slightly by doing something like 
```javascript
  const [value, loading, error] = useCollection(
    db.collection(tokenInput || 'noValue'),
    {
      snapshotListenOptions: { includeMetadataChanges: true },
    },
  );
```
However, I agree that that feels clunky and like what you have more.

Reply:
Your solution is definitely a little cleaner, but yes, as long as there are no strong opinions here, I feel good sticking with what we've written up.